relay-metrics/src/aggregation.rs

```diff
@@ -1055,6 +1080,8 @@ impl Aggregator {
         let timestamp = key.timestamp;
         let project_key = key.project_key;
 
+        let key = key.validate_strings().map_err(|()| AggregateMetricsError)?;
```

Comment:
this will fail the entire batch. we have two options:

* map the error to `return Ok(())`
* make the error type an enum with multiple variants, and have matching logic in `merge_all`